src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +81,12 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(isWomanValid(femaleAge))
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
Here, I think, you can also combine this 
```suggestion
               .flatMap(person -> person.getCats().stream())
```